### ISSUE TITLE
RegulatedNoise in general

### ISSUE THREAD

**Issue by Duke-Jones** [Duke-Jones/RegulatedNoise#57]
Hi zericco,
I open this discussion as a general reconciliation thread.

I'm glad, I'm no longer alone with this project. You've added 
very useful structures (confessedly I had no deep look on the details until now )
and RN seems to be much faster than before. :-) 
On the other hand it's a weird feeling for me to see someone put many things in the trashcan
(and some of them came from me - like the Milkyway) and create other things for it - but that's life. 
( Ok, your solution seems to be better :-) ) 

As I told, next time I will not spend so much time for RN  as in the past,
but of course I will develop the one or other thing to enhance RN. But before I start
the next thing, I think it's useful to wait until you've finished the main part of your 
clean up, because it make's no sense for me to put something in RN and 
it will be deleted on next merging or overwritten with another functionality (even if it's better).
What are you planning next time and what do you think when you're finished with it.

Also we should use this thread to discuss ideas and planned features - so both of us are clued up on them.

I suggest to create a issue for each wanted feature.
If one of us decides to handle one of those issues he should 
mark it as "taken" or "in progress". So it's clear if anyone already 
is working on it.

So far for the moment ... :-P

What do you think about my suggestions ?

PS: I like to share the incoming donations (if the next is coming some time) - but now I'm the 
first time on the recipients side and I can tell you - don't expect too much.

**Comment by WiredSharp** [comment]
Hi DJ, you are totally right, i may be too intrusive in the code and perhaps, from now on we should talk together before heavy rewriting. I am currently working on a brand new domain model which allow us a faster an deasier way of retrieving data and most of all, isolated from UI concerns. My work is isolated in a new assembly and have no impact on current project. As i have alread told you, i target a clean domain model first, then a new GUI in WPF.
Before being able to work concurrently, we need to split the main form in order to avoid conflicts. But, in order to split it, we need stable bases.
I am very sorry if i hurt with my modifications, it was not the right way, but i do not find a way to discuss easily as we have our own rythm each other.
i will use issue for notification of upcoming work and let you add your comments before merge.
It is very nice to you to share donations.

**Comment by Duke-Jones** [comment]
> "...way of retrieving data..."

with "retrieving" you mean retrieving data from the internal RN database ?
If I understand this correct you release the data from RN and put it in 
a "stand alone" assembly like the Brainerous component ? ist this correct ?
I'm very nosy for it - sounds like a kind of selfmade sql-database

Do you have an assessment how long do you need for those steps 
- domain model
- splitting GUI

**Comment by Duke-Jones** [comment]
oh, i forgot something which could be important :

someone asked me to add a setting where the pilot only gets access to
the data of those systems where he already has been. 
It's in my opinion a good idea because many peoply don't like 
to get all the data from other players and wan't to discover
the systems by them self. 
It's like the EDDN marketdata - you CAN use it with RN but you DON'T NEED to use it.
It would be great if RN provides the same functionality for the system data .

basic approach:
to make all data of a system accessible not before the user docks 
first time on a station in this system - it's like a data transfer from station.

I thought, I tell you better now about this idea before it later makes trouble.

**Comment by WiredSharp** [comment]
yes, i was thinking of a notion of perimeter which handle restrictions like distance to system, age of data, and so on, we can also consider visited data. The problem is that we can't retrieve stations visited by a given pilot if logs are not available.

**Comment by Duke-Jones** [comment]
idea: on first start RN can offer one or more of the following options to clear the access if
the "restricted mode" is selected:
- get systems from the Commander's Log
-  get systems from the selfcollected market data (EDDN data is identifiably and can be excluded)
- get all systems within n lightyears from current location
- manual input of single systems

**Comment by WiredSharp** [comment]
yes, that's an idea.

**Comment by sutex** [comment]
Excellent the left hand and the right hand are talking lol should have no problems with the shoelaces now :) 
As for re -inventing the wheel , there are two ideas  label enhancements one is the AER interface and the other is the rare trading program, both have merit. whether or not is incorporated is up to the DEV , but its there to be looked at , before the new wheel design rolls off the drawing board :)

Also , like to mention that one of the great benefits RN has over all others is it local database , the sharing/restricting of this  data to friends and/or groups ( maybe player guild structures later in game) is something to consider, that,  and the web server function , to allow Full screen mode as a lot of players are running ED with DSR which requires it , hence the AER interface suggestion , that's it cya :)

**Comment by Duke-Jones** [comment]
hi sutex, thanks for your continued support of RN

hi zericco,
As you maybe perceived I detached you as collaborator
for this repository. You have very good ideas and I must concede
that you're very good programmer, but I feel we have too different views 
of "whats important for the moment". 
There are too much problems with consisting functions with your 
reconstruction at the moment and I think over to make a rollback.

I hope you're continuing your version of RN and maybe in near future
your version runs stable with all functionality and maybe I also decide for myself
to use your version instead of my version.
But at the moment I will continue my personal version as my own vision.

I can only underline that RN is not my software - I have no rights on it.
It is open source and everyone is (for my part) invited to create his own version.

Sorry for this,  please don't be annoyed,
... in any case good luck with your own version....

Duke

**Comment by WiredSharp** [comment]
Ok, no problem.
